scripts/zones/Bibiki_Bay/mobs/Dalham.lua

```diff
@@ -7,13 +7,44 @@ local entity = {}
 
 entity.onMobInitialize = function(mob)
     mob:setMobMod(xi.mobMod.IDLE_DESPAWN, 180)
+end
+
+entity.onMobSpawn = function(mob)
+    mob:setMobMod(xi.mobMod.ADD_EFFECT, 1)
+    mob:setMod(xi.mod.SLEEPRES, 90)
+    mob:addImmunity(xi.immunity.LIGHT_SLEEP)
+    mob:addImmunity(xi.immunity.SILENCE)
+    mob:addImmunity(xi.immunity.TERROR)
     mob:setMod(xi.mod.WATER_ABSORB, 100)
 end
 
+entity.onAdditionalEffect = function(mob, target, damage)
+    return xi.mob.onAddEffect(mob, target, damage, xi.mob.ae.ENWATER, { chance = 100 })
+end
+
 entity.onMobFight = function(mob, target)
-    -- "Increases swings per attack round as its HP lowers. At 15-20% swings like Charybdis giving the illusion of Hundred Fists."
-    local swings = 1 + math.floor((100 - mob:getHPP()) / 18)
-    mob:setMobMod(xi.mobMod.MULTI_HIT, swings)
+    if mob:getHPP() < 75 and mob:getLocalVar('multihitControl') == 0 then
+        mob:setLocalVar('multihitControl', 1)
+        -- set DOUBLE_ATTACK to 100
+        -- cannot use MULTI_HIT because that is just upto X hits
+        mob:setMod(xi.mod.DOUBLE_ATTACK, 100)
+        -- inject packet for 2hr animation
+        mob:injectActionPacket(mob:getID(), 11, 437, 0, 0x18, 0, 0, 626)
+    elseif mob:getHPP() < 50 and mob:getLocalVar('multihitControl') == 1 then
+        mob:setLocalVar('multihitControl', 2)
+        -- set TRIPLE_ATTACK to 100
+        mob:setMod(xi.mod.DOUBLE_ATTACK, 0)
+        mob:setMod(xi.mod.TRIPLE_ATTACK, 100)
+        -- inject packet for 2hr animation
+        mob:injectActionPacket(mob:getID(), 11, 437, 0, 0x18, 0, 0, 626)
+    elseif mob:getHPP() < 25 and mob:getLocalVar('multihitControl') == 2 then
+        mob:setLocalVar('multihitControl', 3)
+        -- set QUAD_ATTACK to 100
+        mob:setMod(xi.mod.TRIPLE_ATTACK, 0)
+        mob:setMod(xi.mod.QUAD_ATTACK, 100)
+        -- inject packet for 2hr animation
+        mob:injectActionPacket(mob:getID(), 11, 437, 0, 0x18, 0, 0, 626)
+    end
```

Comment:
We could store both the local variable and the mob HPP into alocal variable, so we dont call them potentially 3 times per tick.

Reply:
> We could store both the mob_local_variable and the mob_HPP into local variables before the if/elseif chain, so we dont call them potentially 3 times per tick.

Thanks, I have updated the PR!